examples/DynaSwing/description/DynaSwingExamples.tex

```diff
@@ -607,6 +618,120 @@ \section{Results}
   \caption{Active and reactive powers in pu}
 \end{figure}
 
+\section{BESS}
+
+\subsection{Test Case Description}
+The BESS is connected to an infinite bus through a line.
+
+\begin{figure}[H]
+\center
+  \includegraphics[width=0.55\textwidth]{BESS.png}
+  \caption{Test case diagram}
+\end{figure}
+
+
+\subsubsection{Initial Conditions and parameters}
+The data of the test case were compared to the one of the article
+[P. Pourbeik and J. K. Petter, “Modeling and validation of battery energy storage systems
+using simple generic models for power system stability studies”, CIGRE Science and
+Engineering, October 2017, pp. 63-72.]
+
+The base used in the article is in $SNom =6\ MVA$
+The initial values of P0Pu et Q0Pu are in base SnRef and in receptor convention.
+
+\begin{center}
+\begin{tabular}{|c|c|c|c|c|c|}
+  \hline
+ P0Pu & Q0PU & U0Pu & $\Theta$ (rad) & RPu & XPu\\
+  \hline
+ -0.03 & 0 & 1 & 0.00000144621 & 0 & 0\\
+  \hline
+\end{tabular}
+\end{center}
+
+RPu and XPu are set to 0 in the article, considering that we are controling the output at the injector level (not after the line of BESS).
+Numerically, XPu causes simulation failure at initialization, it is thus set to $10^{-6}$.
```

Comment:
Don't we have a division by zero and thus we should protect this in the model implementation with some if structure instead of using an epsilon value

Reply:
We use XPu in the line model and the REPC-A model but I didn't find any division by 0, these are the equations : 

line :
parameter Types.ComplexImpedancePu ZPu(re = RPu, im = XPu) "Line impedance";
parameter Types.ComplexAdmittancePu YPu(re = GPu, im = BPu) "Line half-admittance";
 ZPu * (terminal2.i - YPu * terminal2.V) = terminal2.V - terminal1.V;
 ZPu * (terminal1.i - YPu * terminal1.V) = terminal1.V - terminal2.V;

REPC-A lineFropCompensation : U1Pu = ComplexMath.'abs'(u2Pu + iPu * Complex(RcPu, XcPu));

Maybe I am missing something, I will look at the logs.

Reply:
Yes, it's what I was supposing too. Perhaps it is then manipulated by the compiler to end up in a form like V1 - V2 / Z. 
Perhaps you can give it a try in the following form:

if ZPu != 0 then currentEq else terminal1.V = terminal2.V;

Reply:
To be resolved in another issue